### DISCUSSION TITLE
Redirect between previews in Monorepo

### DISCUSSION THREAD

**Discussion by DonHuskini** [vercel/community#3487] [Help] [answered]
### Summary

Hi, in our company we use Turborepo to deploy 2 apps and we redirect users from one to the other. It works nice in Production but we couldn't solve it for Preview because the URLs generated for each App are too different.

App 1 URL: `project-1-name-git-fix-delivery-popup-showing-when-item-al-268b38.preview.company.com`
App 2 URL: `project-2-name-git-fix-delivery-popup-showing-when-item-a-0d0c28.preview.company.com`

The branch name is `fix-delivery-popup-showing-when-item-already-exists` and you can see it crops differently in both URLs. Also, I don't know how the last key is genarated `268b38` and `0d0c28`.

I need to be able to generate the correct Preview URL of App 2 in App 1 so I can redirect users from one to the other. Turborepo is setup to deploy both apps on every new commit and and branch.

### Steps to Reproduce

1. Deploy 2 different apps of the same monorepo in preview
2. Try to redirect from one to the other

**Comment by amyegan** [comment, marked as answer]
Hey @gabrielmlinassi. Each deployment is given its own URL with a unique hash that is randomly generated. However, there are other automatically generated URLs that are more predictable. 

When a deploying from a git branch, the latest deployment from that branch will be assigned a [branch-based URL](https://vercel.com/docs/concepts/deployments/generated-urls#url-with-git-branch) in the form of `<project-name>-git-<branch-name>-<scope-slug>.vercel.app`. You can combine the `VERCEL_GIT_COMMIT_REF` environment variable with the project name and account slug to build the URL and redirect to the other project's preview deployments.

CLI deployments are also assigned a similarly predictable URL. You can learn more about generated URLs at https://vercel.com/docs/concepts/deployments/generated-urls#

**Reply by hussamkhatib** [reply]
From what I have observed the **predictable** preview deployment URL appears only to the latest preview build. This might be desirable when deploying using git branch, but a deal breaker for CLI deployments. For example using CLI as part of CI/CD pipeline which gets triggered on pull request, the predicable URL will be pointing to the latest PR/commit.
